c2rust-transpile/src/translator/assembly.rs

```diff
@@ -262,13 +391,15 @@ fn asm_is_att_syntax(asm: &str) -> bool {
     } else if asm.contains("word ptr") {
         false
     } else {
-        asm.contains('$')
+        // Guess based on sigils used in AT&T assembly. This would be more
+        // robust if it stripped comments first.
+        asm.contains('$') || asm.contains('%') || asm.contains('(')
```

Comment:
I'm a bit concerned that '(' is too broad and will end up with false positives.

Reply:
'(' is intended to pick out `disp(base,idx,scale)` address calculations in AT&T syntax. Is there anywhere it can appear in Intel syntax? The only places I can think of are comments and macros. It's not hard to remove comments, but parsing macros seems like it's more trouble than we should bother with.

Reply:
We now remove comments before searching for these sigils.